google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/QueryResult.java

```diff
@@ -16,71 +16,35 @@
 
 package com.google.cloud.bigquery;
 
-import com.google.cloud.PageImpl;
-import com.google.common.base.MoreObjects;
-import com.google.common.collect.ImmutableList;
+import static com.google.common.base.Preconditions.checkNotNull;
 
+import com.google.api.gax.paging.Page;
+import com.google.common.base.Function;
+import com.google.common.base.MoreObjects;
+import com.google.common.collect.Iterables;
 import java.util.Objects;
 
-public class QueryResult extends PageImpl<FieldValueList> {
+// TODO(pongad): Serializable ?
```

Comment:
@tswast I don't completely understand what being `Serializable` requires. Does this class needs to be changed in some way?

I haven't seen any errors, but thought I should actually think about it instead of blindly typing :laughing: